### PR TITLE
Fs prospect upload validation w table 58

### PR DESCRIPTION
**Pull request by mcgraths7**
This PR covers issues 57 & 58

**What it does**

1. Validates presence of email address in CSV file
2. Allows user to specify which header is associated with which column via a select input
3. Validates uniqueness of email addresses
4. Displays message snackbars on success, or when there is a problem
5. Displays the prospect data in a table so the user can verify it is correct
6. Integrates with new Prospect and User models on backend

**What to look out for**
1. New "BasicTable" component, which isn't very extendable at this time, but can be adjusted in the future if needed
2. New utility functions - generateUniqueId & validateEmail
3. There are still some dangling unused components which might need to be cleaned up in a future PR (ComboButton, TableSelect
4. Currently, there is no functionality in the drawer, but it seemed like a better experience to maintain it rather than have a new page structure
5. I chose the select boxes because it seemed the easiest to implement on a first pass.

**Known issues**
1. The overall styling isn't quite there yet.
2. Some styles are scattered, need to consolidate
3. Select boxes aren't embedded in the table. Determining whether to just make a specific UploadTable component for this one task
4. There's currently no tracking of duplicates as of the latest commit, but will be resolved in a later issue

### CODE REVIEW

**Review by Patton-L — COMMENTED** [review]
Backend parts look good. The name of Prospect field is different from my branch, but we can figure that out during merge.

**Review by Patton-L — APPROVED** [review]

**Comment by mcgraths7** [comment]
> Backend parts look good. The name of Prospect field is different from my branch, but we can figure that out during merge.

I'll defer to yours and reformat the frontend to match when the time comes. Might let my PR sit until I can rebase with your changes

**Comment by Patton-L** [comment]
> > Backend parts look good. The name of Prospect field is different from my branch, but we can figure that out during merge.
> 
> I'll defer to yours and reformat the frontend to match when the time comes. Might let my PR sit until I can rebase with your changes

Got it

File: server/api/prospect_handler.py
Comment (skassam21) [inline server/api/prospect_handler.py:55]: This should be coming from the JWT identity (for security purposes, or anyone can pass any owner email)

File: server/api/prospect_handler.py
Comment (skassam21) [inline server/api/prospect_handler.py]: let's remove the commented out code

File: server/db/model.py
Comment (skassam21) [inline server/db/model.py]: I don't think we need campaigns, steps, last_contacted right now - so let's just remove it

File: server/db/model.py
Comment (skassam21) [inline server/db/model.py:262]: self.prospects doesn't exist anymore, right @Patton-L - so in this case, I would revert back to how you did it before Steven, but just add the logic to this file

File: server/api/prospect_handler.py
Comment (mcgraths7) [reply server/api/prospect_handler.py:55]: Resolved in PR #66

File: server/api/prospect_handler.py
Comment (mcgraths7) [reply server/api/prospect_handler.py]: Removed

File: server/db/model.py
Comment (mcgraths7) [reply server/db/model.py]: Removed the unused fields

File: server/db/model.py
Comment (mcgraths7) [reply server/db/model.py:262]: Looks like "prospects" is a property on User model. Line 267

**Review by skassam21 — APPROVED** [review]
Let's pull the changes from the other PR after merging, then you can merge this one

**Pull request merged**
